javascript/ql/src/semmle/javascript/frameworks/Electron.qll

```diff
@@ -73,187 +73,109 @@ module Electron {
+    class MainProcess extends Process {
+      MainProcess() { this = main() }
     }
 
     /**
-     * A synchronous IPC message sent directly from a process.
+     * An instance of the renderer process of an Electron app.
      */
-    class SyncDirectMessage extends DirectMessage {
-      SyncDirectMessage() { isSync = true }
-
-      /** Gets the data flow node holding the reply to the message. */
-      DataFlow::Node getReply() { result = mc }
+    class RendererProcess extends Process {
+      RendererProcess() { this = renderer() }
     }
 
     /**
-     * An asynchronous IPC reply sent from within an IPC callback.
+     * The `sender` property of the event in an IPC event handler.
+     * This sender is used to send a response back from the main process to the renderer.
      */
-    class AsyncReplyMessage extends Message {
-      Callback callback;
-      DataFlow::Node channel;
-
-      AsyncReplyMessage() {
-        exists(DataFlow::MethodCallNode mc |
-          mc = callback.getParameter(0).getAPropertyRead("sender").getAMemberCall("send") and
-          this = mc.getArgument(1) and
-          channel = mc.getArgument(0)
+    class ProcessSender extends Process {
+      ProcessSender() {
+        exists(IPCSendRegistration reg | reg.getEmitter() instanceof MainProcess |
+          this = reg.getABoundCallbackParameter(1, 0).getAPropertyRead("sender")
         )
       }
-
-      override Process getProcess() { result = callback.getProcess() }
-
-      override string getChannelName() { result = channel.getStringValue() }
     }
 
     /**
-     * A synchronous IPC reply sent from within an IPC callback.
+     * A registration of an Electron IPC event handler.
+     * Does mostly the same as an EventEmitter event handler,
+     * except that values can be returned through the `event.returnValue` property.
      */
-    class SyncReplyMessage extends Message {
-      Callback callback;
+    class IPCSendRegistration extends EventEmitter::EventRegistration::Range,
+      DataFlow::MethodCallNode {
+      override Process emitter;
 
-      SyncReplyMessage() {
-        this = callback.getParameter(0).getAPropertyWrite("returnValue").getRhs()
-      }
+      IPCSendRegistration() { this = emitter.ref().getAMethodCall(EventEmitter::on()) }
 
-      override Process getProcess() { result = callback.getProcess() }
+      override DataFlow::Node getAReturnedValue() {
+        result = this.getABoundCallbackParameter(1, 0).getAPropertyWrite("returnValue").getRhs()
+      }
 
-      override string getChannelName() { result = callback.getChannelName() }
+      override predicate canReturnTo(EventEmitter::EventDispatch dispatch) {
+        dispatch.(IPCDispatch).getCalleeName() = "sendSync"
+      }
     }
 
     /**
-     * An asynchronous Electron IPC message sent from the main process via a `webContents` object.
+     * A dispatch of an IPC event.
+     * An IPC event is sent from the Renderer to the Main process.
```

Comment:
You capitalise "Renderer" and "Main" here, but not elsewhere. I don't mind either way, but it should be consistent.